ElectronicObserver/Window/Tools/ExpeditionCheck/ExpeditionCheckView.xaml

```diff
@@ -0,0 +1,159 @@
+﻿<common:UserControlBase
+    x:Class="ElectronicObserver.Window.Tools.ExpeditionCheck.ExpeditionCheckView"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:b="http://schemas.microsoft.com/xaml/behaviors"
+    xmlns:common="clr-namespace:ElectronicObserver.Common"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:local="clr-namespace:ElectronicObserver.Window.Tools.ExpeditionCheck"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    xmlns:persistentColumns="clr-namespace:ElectronicObserver.Behaviors.PersistentColumns"
+    Width="auto"
+    Height="auto"
+    d:DataContext="{d:DesignInstance local:ExpeditionCheckViewModel}"
+    d:DesignHeight="450"
+    d:DesignWidth="800"
+    x:TypeArguments="local:ExpeditionCheckViewModel"
+    mc:Ignorable="d"
+    >
+
+    <UserControl.Resources>
+        <local:ExpeditionCheckViewBindingProxy x:Key="Proxy" DataContext="{Binding}" />
+    </UserControl.Resources>
+
+    <Grid Visibility="Visible">
+
+        <Grid.Resources>
+            <Style TargetType="TextBlock">
+                <Setter Property="Margin" Value="4 0" />
+                <Setter Property="VerticalAlignment" Value="Center" />
+            </Style>
+        </Grid.Resources>
+
+        <DataGrid
+            AutoGenerateColumns="False"
+            ColumnWidth="90"
+            HeadersVisibility="Column"
+            IsReadOnly="True"
+            ItemsSource="{Binding Rows}"
+            >
+
+            <b:Interaction.Behaviors>
+                <persistentColumns:PersistentColumnsBehavior ColumnProperties="{Binding ColumnProperties}" SortDescriptions="{Binding SortDescriptions}" />
+            </b:Interaction.Behaviors>
+
+            <DataGrid.ColumnHeaderStyle>
+                <Style BasedOn="{StaticResource DefaultDataGridColumnHeaderStyle}" TargetType="{x:Type DataGridColumnHeader}">
+                    <Setter Property="Foreground" Value="{Binding DataContext.FontBrush, Source={StaticResource Proxy}}" />
+                    <Setter Property="FontFamily" Value="{Binding DataContext.Font, Source={StaticResource Proxy}}" />
+                    <Setter Property="FontSize" Value="{Binding DataContext.FontSize, Source={StaticResource Proxy}}" />
+                </Style>
+            </DataGrid.ColumnHeaderStyle>
+
+            <DataGrid.RowStyle>
+                <Style TargetType="DataGridRow">
+                    <Setter Property="Background" Value="{Binding Background}" />
+                </Style>
+            </DataGrid.RowStyle>
+
+            <DataGrid.Columns>
+                <DataGridTextColumn
+                    Width="Auto"
+                    Binding="{Binding IdDisplay}"
```

Comment:
The current sorting behavior doesn't match the old version. You need to add `SortMemberPath="SortId"` and then in ExpeditionCheckRow define the SortId property.

In the old code the compare is by MapAreaId and then by MissionId. So I think `SortId = MapAreaId * 1000 + MissionId` should do the trick.